src/commands/sync.ts

```diff
@@ -0,0 +1,214 @@
+import File from '../core/File'
+
+import { arrayPartition, MissingWorkspaceError } from '../helpers'
+
+const enum GroupType {
+  configs = 'configs',
+  content = 'content',
+  specs = 'specs',
+  emails = 'emails',
+  themes = 'themes',
+  others = 'others',
+}
+
+interface ISyncArguments {
+  workspace: string
+  disableSSLVerification?: boolean
+  ignoreSpecs?: boolean
+  skipPath?: string[]
+  verbose?: boolean
+  watch?: boolean
+}
+
+interface ISyncOptions {
+  path?: string
+}
+
+type FileGroups = Partial<Record<GroupType, File[]>>
+
+function buildFilesObject(fileGroups: FileGroups, files: File[], filterBy: GroupType): File[] {
+  // each time this function runs we create a new property based on filterBy and remove it from files
+  // this way we never miss anything from files
+
+  if (filterBy === GroupType.configs) {
+    // find files in the root directory and add them to the configs group
+    const { matched, mismatched } = arrayPartition(files, (file) => !/\//.test(file.resource.path || ''))
+    fileGroups.configs = matched
+    files = mismatched
+
+    return files
+  }
+
+  // find files with paths starting with filterBy and add them to their group
+  const { matched, mismatched } = arrayPartition(files, (file) => file.resource.path?.startsWith(filterBy))
+  fileGroups[filterBy] = matched
+
+  return mismatched
+}
+
+async function Sync(workspace: Workspace, args: ISyncArguments, options: ISyncOptions = {}): Promise<void> {
+  console.log(`Syncing ${workspace.name}:\n`)
+
+  const verbose = (...verboseArgs) => args.verbose && console.debug(...verboseArgs)
+
+  const spinner = ora({
+    prefixText: `Syncing ${options.path || 'all workspace files'}...`,
+    text: `reading files...`,
+  }).start()
+
+  const client = new RestClient(workspace.config, workspace.name)
+
+  try {
+    // scan upstream workspace files
+    const upstreamFiles = await client.getAllFiles({ fields: 'path,checksum' })
+
+    // scan workspace directories and build list of files
+    const files = await workspace.scan()
+
+    spinner.text = `Found ${files.length} files locally and ${upstreamFiles.length} files upstream`
+    spinner.succeed().start()
+
+    // assign files to their groups
+    const fileGroups: FileGroups = {}
+    let remainingFiles = [...files]
+    remainingFiles = buildFilesObject(fileGroups, remainingFiles, GroupType.configs)
+    remainingFiles = buildFilesObject(fileGroups, remainingFiles, GroupType.content)
+    remainingFiles = buildFilesObject(fileGroups, remainingFiles, GroupType.specs)
+    remainingFiles = buildFilesObject(fileGroups, remainingFiles, GroupType.emails)
+    remainingFiles = buildFilesObject(fileGroups, remainingFiles, GroupType.themes)
+
+    if (files.length > 0) {
+      fileGroups[GroupType.others] = remainingFiles
+    }
+
+    const updatedPaths: string[] = []
+    const deletedPaths: string[] = []
+
+    for (const [group, files] of Object.entries(fileGroups)) {
```

Comment:
I'd really liek to rename the `files` to `innerFiles` or something since it collides with the above `files` constant. Just for cognitive load's sake.

Reply:
Updated `files` to `allFiles` instead to make it more clear :)